crates/api_common/src/request.rs

```diff
@@ -140,6 +145,59 @@ pub(crate) async fn fetch_pictrs(
   }
 }
 
+/// Purges an image from pictrs
+/// Note: This should often be coerced from a Result to .ok() in order to fail softly, because:
+/// - It might fail due to image being not local
+/// - It might not be an image
+/// - Pictrs might not be set up
+pub async fn purge_image_from_pictrs(
+  client: &ClientWithMiddleware,
+  settings: &Settings,
+  image_url: &Url,
+) -> Result<(), LemmyError> {
+  if let Some(pictrs_config) = settings.pictrs_config.to_owned() {
```

Comment:
Use the pictrs_config function you defined in lemmy_routes (after moving it to lemmy_utils).